waku/v2/protocol/rln/group_manager/root_tracker.go

```diff
@@ -125,29 +132,50 @@ func (m *MerkleRootTracker) pushRoot(blockNumber uint64, root [32]byte) {
 	if len(m.merkleRootBuffer) > maxBufferSize {
 		m.merkleRootBuffer = m.merkleRootBuffer[1:]
 	}
-
 }
 
+// Roots return the list of valid merkle roots
 func (m *MerkleRootTracker) Roots() []rln.MerkleNode {
 	m.RLock()
 	defer m.RUnlock()
 
 	result := make([]rln.MerkleNode, len(m.validMerkleRoots))
 	for i := range m.validMerkleRoots {
-		result[i] = m.validMerkleRoots[i].root
+		result[i] = m.validMerkleRoots[i].Root
 	}
 
 	return result
 }
 
+// Buffer is used as a repository of older merkle roots that although
+// they were valid once, they have left the acceptable window of
+// merkle roots. We keep track of them in case a chain fork occurs
+// and we need to restore the valid merkle roots to a previous point
+// of time
```

Comment:
I am wondering if the following scenario would cause an issue.
Since we are not persisting buffer, what if a node only persists latest roots and restarts and there is a chain fork in between restart. how will the node recover its validRoots? 
Should we persist buffer or identify if there was a chain fork after restart and resync from deployed block? Not sure which is costly

Reply:
cc: @rymnc